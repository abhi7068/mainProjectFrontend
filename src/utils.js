import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { createStore, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';
import { createMemoryHistory } from 'history';
import thunk from 'redux-thunk';
import PropTypes from 'prop-types';
import reducer from '../src/store/reducers/rootReducer';
import { Router } from 'react-router-dom';

function render(
  ui,
  {
    initialState,
    store = createStore(reducer, initialState, applyMiddleware(thunk)),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    const history = createMemoryHistory();
    return (
      <Provider store={store}>
        <Router history={history}>{children}</Router>
      </Provider>
    );
  }
  Wrapper.propTypes = {
    children: PropTypes.elementType,
  };
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}
export * from '@testing-library/react';
export { render };
