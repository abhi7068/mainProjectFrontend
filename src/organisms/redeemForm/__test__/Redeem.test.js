import React from 'react';
import ReactDOM from 'react-dom';
import Redeem from '../index';
import { render, fireEvent } from '../../../utils';
import { Provider } from 'react-redux';
import reducer from '../../../store/reducers/rootReducer';
import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { BrowserRouter } from 'react-router-dom';
const store = createStore(reducer, applyMiddleware(thunk));

it('it renders', () => {
  render(<Redeem />);
});

it('should take a snapshot', () => {
  const { asFragment } = render(<Redeem />);
  expect(asFragment(render(<Redeem />))).toMatchSnapshot();
});

it('renders with a expected length', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <Provider store={store}>
      <BrowserRouter>
        <Redeem />
      </BrowserRouter>
    </Provider>,
    div
  );
  expect(div.querySelectorAll('input')).toHaveLength(1);
});
it('it renders with correct text', () => {
  const { getByTestId } = render(<Redeem />);
  expect(getByTestId('button')).toHaveTextContent('Redeem');
});
it('renders with expected attribute', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <Provider store={store}>
      <BrowserRouter>
        <Redeem />
      </BrowserRouter>
    </Provider>,
    div
  );
  expect(div.querySelector('input')).toHaveAttribute('type', 'text');
});
it('to be true', () => {
  const { getByTestId } = render(<Redeem />);
  expect(fireEvent.click(getByTestId('button'))).toBe(true);
});
