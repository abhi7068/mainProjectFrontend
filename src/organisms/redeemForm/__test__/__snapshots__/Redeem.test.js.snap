// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="MuiPaper-root MuiCard-root SendForm-root-174 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        class="MuiCardHeader-root"
      >
        <div
          class="MuiCardHeader-content"
        >
          <span
            class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
          >
            Redeem You Gift
          </span>
        </div>
      </div>
      <form>
        <div
          class="MuiCardContent-root"
        >
          <div>
            <legend
              class="MuiFormLabel-root"
              data-testid="label"
            >
              Coupon Code
            </legend>
          </div>
          <br />
          <div
            class="MuiFormControl-root MuiTextField-root"
            style="width: 225px;"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
              data-shrink="false"
              for="outlined-basic"
              id="outlined-basic-label"
            >
              Redeem Your Gift
              <span
                class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input"
                id="outlined-basic"
                placeholder="Redeem Code"
                required=""
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-301 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-303"
                >
                  <span>
                    Redeem Your Gift *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="MuiCardContent-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
            data-testid="button"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              Redeem
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
